### PR TITLE
fix!: fix saas related issues

### PR DESCRIPTION
**Pull request by pweyck**

### CODE REVIEW

File: src/page-objects/administration/Categories.ts [hunk 1]
```diff
@@ -14,7 +14,15 @@ export class Categories implements PageObject {
...
+
+  /**
+   * @deprecated Please don't depend on the default categories
+   */
   public readonly homeCategory: Locator;
+
+  /**
+   * @deprecated Please don't depend on the default categories
+   */
   public readonly homeCategoryContextButton: Locator;
```
Comment (pweyck) [inline src/page-objects/administration/Categories.ts]: Maybe it's better to remove this and break it, because we don't do this anyway

Comment (pweyck) [reply src/page-objects/administration/Categories.ts]: I've removed the home stuff

File: src/page-objects/administration/Categories.ts [hunk 2]
```diff
@@ -14,8 +14,6 @@ export class Categories implements PageObject {
    * Category tree
    */
   public readonly categoryTree: Locator;
-  public readonly homeCategory: Locator;
```
Comment (pweyck) [inline src/page-objects/administration/Categories.ts:17]: saas and potentially many other systems do not have the home category. We should always create all data that we expect to be in the system

**Review by vanpham-sw — APPROVED** [review]

File: src/page-objects/administration/Categories.ts [hunk 3]
```diff
@@ -51,8 +49,7 @@ export class Categories implements PageObject {
     this.addLandingPageButton = this.landingPageArea.getByText('Add landing page');
     this.landingPageItems = this.landingPageArea.locator('.sw-tree-item__label');
     this.categoryTree = page.locator('.sw-category-tree');
-    this.homeCategory = this.categoryTree.locator('.sw-tree-item__element').filter({ hasText: 'Home' });
```
Comment (frobel) [inline src/page-objects/administration/Categories.ts:54]: if we remove that, we have to adjust the a platform test or?

File: tests/TestDataService/TestDataService.spec.ts [hunk 4]
```diff
@@ -143,9 +141,16 @@ test('Data Service', async ({
     const { data: databaseCategory } = (await categoryResponse.json()) as { data: Category };
     expect(databaseCategory.id).toBe(category.id);
 
-    const systemConfigEntryResponse = await AdminApiContext.get(`./system-config/${systemConfigEntry.id}?_response=detail`);
-    const { data: databaseSystemConfigEntry } = (await systemConfigEntryResponse.json()) as { data: SystemConfig };
-    expect(databaseSystemConfigEntry.id).toBe(systemConfigEntry.id);
+    // eslint-disable-next-line playwright/no-conditional-in-test
+    if (!isSaaSInstance(AdminApiContext)) {
```
Comment (frobel) [inline tests/TestDataService/TestDataService.spec.ts:145]: ow yes, good point

**Review by frobel — APPROVED** [review]

File: src/page-objects/administration/Categories.ts (hunk 3 above)
Comment (frobel) [reply src/page-objects/administration/Categories.ts:54]: @vanpham-sw you aware of it?

Comment (vanpham-sw) [reply src/page-objects/administration/Categories.ts:54]: Yes, we have to adjust the platform test:
From this:
`await ShopAdmin.attemptsTo(CreateLinkTypeCategory(categoryData, categoryCustomizableLinkData));`
to:
`await ShopAdmin.attemptsTo(CreateLinkTypeCategory(categoryData, categoryCustomizableLinkData, 'Home'));`

I can create a MR, but maybe there is a better way?

Comment (pweyck) [reply src/page-objects/administration/Categories.ts:54]: I plan to do that :)

**Pull request merged**